addons/l10n_tr_nilvera_einvoice/models/account_move.py

```diff
@@ -0,0 +1,253 @@
+import uuid
+from markupsafe import Markup
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _name = 'account.move'
+    _inherit = ['account.move']
+
+    def _default_l10n_tr_nilvera_uuid(self):
+        return str(uuid.uuid4())
+
+    l10n_tr_nilvera_einvoice_xml_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        string="Nilvera Attachment",
+        compute=lambda self: self._compute_linked_attachment_id('l10n_tr_nilvera_einvoice_xml_id',
+                                                                'l10n_tr_nilvera_einvoice_xml_file'),
+        depends=['l10n_tr_nilvera_einvoice_xml_file']
+    )
+    l10n_tr_nilvera_einvoice_xml_file = fields.Binary(
+        attachment=True,
+        string="Nilvera File",
+        copy=False,
+    )
+    l10n_tr_nilvera_uuid = fields.Char(
+        string='Nilvera Document UUID',
+        copy=False,
+        readonly=True,
+        default=_default_l10n_tr_nilvera_uuid,
+        help="Universally unique identifier of the Invoice",
+    )
+    l10n_tr_nilvera_send_status = fields.Selection(
+        selection=[
+            ('error', 'Error (check chatter)'),
+            ('not_sent', 'Not sent'),
+            ('sent', 'Sent and waiting response'),
+            ('succeed', 'Successful'),
+            ('waiting', 'Waiting'),
+            ('unknown', 'Unknown'),
+        ],
+        string='Nilvera Status',
+        readonly=True,
+        copy=False,
+        default='not_sent',
+    )
+
+    @api.model
+    def _get_ubl_cii_builder_from_xml_tree(self, tree):
+        customization_id = tree.find('{*}CustomizationID')
+        if customization_id is not None and 'TR1.2' in customization_id.text:
+            return self.env['account.edi.xml.ubl.tr']
+        return super()._get_ubl_cii_builder_from_xml_tree(tree)
+
+    def _l10n_tr_nilvera_submit_einvoice(self, xml_file, customer_alias):
+        self._l10n_tr_nilvera_submit_document(
+            xml_file=xml_file,
+            endpoint=f"/einvoice/Send/Xml?Alias={customer_alias}",
+            success_message=_("The invoice has been successfully sent to Nilvera as an E-Invoice."),
+            post_series_func=self._l10n_tr_nilvera_post_einvoice_series,
+        )
+
+    def _l10n_tr_nilvera_submit_earchive(self, xml_file):
+        self._l10n_tr_nilvera_submit_document(
+            xml_file=xml_file,
+            endpoint="/earchive/Send/Xml",
+            success_message=_("The invoice has been successfully sent to Nilvera as an E-Arhive."),
+            post_series_func=self._l10n_tr_nilvera_post_earchive_series,
+        )
+
+    def _l10n_tr_nilvera_submit_document(self, xml_file, endpoint, success_message, post_series_func=None):
```

Comment:
There is an XSS problem here: in case of success (**200**), the `success_message` is marked-up as is. However, that message is a function parameter meaning it can potentially be anything. 

You should either remove the parameter completely or escape it before marking it up. I would prefer the first option as I think the value added by this parameter is not worth the trouble, a simple success message like "The invoice has been successfully sent to Nilvera" is sufficient IMO but i'm no PO so..

Reply:
Will check with OLMA when he's back if he's okay with the simpler message

Reply:
Going with the simpler message in the end!